### ISSUE TITLE
Fix the AkkaStreams.guarantee implementation (or test)

### ISSUE THREAD

**Issue by BalmungSan** [neotypes/neotypes#433]
Right now we are ignoring the whole `AkkaStreamsSuite` since the `StreamGuaranteeSpec` started to fail for **AkkaStreams** without any apparent reason _(I believe the implementation always had a bug but we were lucky with the CI tests when we merged the PR)_.

The branch https://github.com/neotypes/neotypes/tree/debug-akka can be used as a starting point to play with the code and try to figure out what is wrong.

-----

**Note:** Another thing that would be good to fix is that right now we are not using the custom `ExecutionContext` provided by **Scalatest** in the creation of the `ActorSystem` used as the `Materializer` for the whole spec.
However, when trying to use it like this:

```scala
implicit val system =
  ActorSystem(name = "QuickStart", defaultExecutionContext = Some(ec))
```

The whole suite fails with some weird error about a timeout waiting for a logger author.

**Comment by zmerr** [comment]
May I ask where this constructor for the `ActorSystem` is coming from? I cannot find it in the current version of [docs](https://doc.akka.io/api/akka/current/akka/actor/ActorSystem.html).

**Comment by zmerr** [comment]
So, how have you initialized the execution context you’ve passed to the actor system?

**Comment by BalmungSan** [comment]
@zmerr I didn't initialized it, is the once that **Scalatest** provides when using an `AsyncFlatSpecLike` which is a single threaded one.

**Comment by BalmungSan** [comment]
Fixed by #440
